Cargo.toml

```diff
@@ -47,6 +47,7 @@ threadpool = "1.8.1"
 rand = "0.8.5"
 humantime-serde = "1.1.1"
 humantime = "2.1.0"
+itertools = "0.13.0"
```

Comment:
don't think this made it into the PR?

Reply:
It did. This is what implements into_group_map_by

Reply:
Oh, it's one of those things that adds magic behind the scenes. Still getting used to that.